crates/core/src/bcmedia/de.rs

```diff
@@ -240,11 +240,41 @@ fn bcmedia_iframe(buf: &[u8]) -> IResult<&[u8], BcMediaIframe> {
         verify(take4, |x| matches!(x, "H264" | "H265")),
     )(buf)?;
     let (buf, payload_size) = le_u32(buf)?;
-    let (buf, _unknown_a) = le_u32(buf)?;
+    let (buf, additional_header_size) = le_u32(buf)?;
     let (buf, microseconds) = le_u32(buf)?;
     let (buf, _unknown_b) = le_u32(buf)?;
-    let (buf, time) = le_u32(buf)?;
-    let (buf, _unknown_c) = le_u32(buf)?;
+    let (buf, time) = if additional_header_size >= 4 {
+        let (buf, time_value) = le_u32(buf)?;
+        (buf, Some(time_value))
+    } else {
+        (buf, None)
+    };
+    let (buf, _unknown_c) = if additional_header_size >= 8 {
+        let (buf, unknown_c) = le_u32(buf)?;
+        (buf, Some(unknown_c))
+    } else {
+        (buf, None)
+    };
+    let (buf, _unknown_d) = if additional_header_size >= 12 {
+        let (buf, unknown_d) = le_u32(buf)?;
+        (buf, Some(unknown_d))
+    } else {
+        (buf, None)
+    };
+    let (buf, _unknown_e) = if additional_header_size >= 16 {
+        let (buf, unknown_e) = le_u32(buf)?;
+        (buf, Some(unknown_e))
+    } else {
+        (buf, None)
+    };
+    let (buf, _unknown_remained) = if additional_header_size > 16 {
+        let remainder = additional_header_size - 16;
+        let (buf, unknown_remained) = take!(buf, remainder)?;
+        (buf, Some(unknown_remained))
+    } else {
+        (buf, None)
+    };
```

Comment:
Why does this have extra `_unknown_[b-e]` fields? Can you just collapse them directly into `unknown_remained`?

Reply:
Ok Done. Was mostly for some testing.

p.s. Nice to hear from you, how are things?